lib/queue.js

```diff
@@ -505,7 +505,14 @@ Queue.prototype.processStalledJob = function(job){
           if(!isMember){
             _this.emit('stalled', job);
             return _this.processJob(job, true);
+          } else {
+            return job.releaseLock(_this.token);
           }
+        }).catch(function(err) {
+          // Any uncaught error will come here. We'll ensure that the job lock is released and rethrow
+          // the error so it bubbles normally
+          job.releaseLock(_this.token);
+          throw err;
         });
```

Comment:
Wouldn't be better to put a finally on the promise returned by job.takeLock() in line 501, so that we always release the lock, and just be sure that even if we release an already released lock we do not throw any exception (which I think will be the standard behaviour inside a finally clause anyway:
```javascript
return job.takeLock(_this.token).then(function(lock){
      if(lock){
        var key = _this.toKey('completed');
        return _this.client.sismemberAsync(key, job.jobId).then(function(isMember){
          if(!isMember){
            _this.emit('stalled', job);
            return _this.processJob(job, true);
          }
        });
      }
    }).finally(function(){
      job.releaseLock(_this.token);
    });
```